src/core/compiler.jl

```diff
@@ -55,19 +61,20 @@ function generate_observe(observation, distribution)
                 vi
             )
         end
-    )
 end
 
 """
-    generate_assume(variable, distribution, syms)
+    generate_assume(variable, distribution, compiler)
 
 Generate an assume expression for parameters `variable` drawn from 
 a distribution or a vector of distributions (`distribution`).
 
 """
-function generate_assume(variable, distribution, syms)
-    return esc(
-        quote
+function generate_assume(variable, distribution, compiler)
+    sym, idcs, csym = @VarName(variable)
+    csym = Symbol(compiler[:name], csym)
+    syms = Symbol[csym, variable]
+    return quote
```

Comment:
This was needed because the `~` macro was expanded separately, so it needed its own `esc`. Now the `~` expressions are expanded in the `model` macro, so only the `model` macro's `esc` is needed.